### ISSUE TITLE
[Question] Why does Wan2.1 script skip first 2 epochs while others do not?

### ISSUE THREAD

**Issue by Dshijie** [yifan123/flow_grpo#202]
### Body:

Thank you for maintaining this impressive repository!

I was studying the training code and noticed a specific logic in the [Wan2.1 script](https://github.com/yifan123/flow_grpo/blob/main/scripts/train_wan2_1.py) which does not appear in the other scripts.

### The Logic:

In the Wan2.1 training loop, there is a check to skip the first two epochs:

```python
# 这里是故意的，因为前两个epoch收集的group size会有bug,经过两个epoch后，group_size稳定成指定的
if epoch < 2:
    continue
```

### My Questions:

1. **What exactly is this "group size bug"?**
I'm not fully understanding the reasoning behind this comment. Why would the group size be unstable or incorrect in the first two epochs? Is this related to the `DistributedKRepeatSampler`, the `DataLoader` state, or something specific to the Wan pipeline?

2. **Why is this inconsistency present?**
I noticed that the [Flux training script](https://github.com/yifan123/flow_grpo/blob/main/scripts/train_flux.py) uses the same `DistributedKRepeatSampler` and similar `DataLoader` settings, but it does not have this skip logic.
    - Does the Flux training pipeline also suffer from this "bug" in the beginning?
    - Or is there a specific reason why Wan2.1 requires this warmup while others do not?

3. **Is it safe to remove this warm-up check?**
If I want to run a standard training job, is it safe to remove this `continue` block?

I would appreciate any insights into the design choice here. Thanks!

**Comment by Jayce-Ping** [comment]
The bug is caused by duplicate preparation for DistributedDataloader. Removing `train_dataloader` will fix it. Specifically,

Change

https://github.com/yifan123/flow_grpo/blob/9745241cf1101407e9b518dd4f10498092150522/scripts/train_wan2_1.py#L591

to

```python
    transformer, optimizer, test_dataloader = accelerator.prepare(transformer, optimizer, test_dataloader)
```

And you can remove the `first 2-epoch skipping code`.

If you want to know more about this part, please refer to Issue #106 and Pull request #107.
